src/cr/cube/dimension.py

```diff
@@ -365,6 +365,23 @@ def apply_transforms(self, dimension_transforms):
             self._dimension_dict, self._dimension_type, dimension_transforms
         )
 
+    @lazyproperty
+    def derived_elements(self):
+        """_Elements object providing access to derived elements.
+
+        Derived elements are insertions computed by zz9.
+        """
+        return self.all_elements.derived_elements
+
+    @lazyproperty
+    def derived_element_ids(self):
+        """tuple of insertion-id for each derived element in this dimension.
+
+        Insertion-ids appear in the order that the derived elements are defined. Can
+        be int or string.
+        """
+        return tuple(s.element_id for s in self.derived_elements)
```

Comment:
`_DerivedElements` already has `.element_ids` inherited from the `_BaseElements` collection class, so minimally the implementation should be `return self.derived_elements.element_ids` rather than reimplementing that here. At the same time, I'd want to have a story convincing myself it's not better for the caller to get those for itself if `.derived_elements` is already public.

I'm thinking either you let folks have `.derived_elements` and they use them how they want (using the standard `_Elements` interface, _OR_ you keep the collection private and publish direct access to what folks want (like `.derived_element_ids`).

I kind of suspect the latter is what's called for here (like `.derived_elements` becomes `._derived_elements`) but I'll need to get my head into the rest of it here to say.